src/main/resources/db/migration/V1_13_1__created_by_auth_user.sql

```diff
@@ -0,0 +1,27 @@
+-- add the new column type that will replace both 'created_by_userid' and 'created_by_user_auth_source'
+alter table referral
+    add column created_by_id text,
+    add constraint fk_created_by_id foreign key (created_by_id) references auth_user;
+
+-- copy over all 'created_by_userid' values to the 'created_by_id' column. these columns hold the same value.
+update referral
+set created_by_id = created_by_userid;
+
+alter table referral
+    alter column created_by_id set not null;
+
+-- create any auth user entities that exist in the referral table but not the auth_user table.
+insert into auth_user (id, auth_source)
+    select created_by_userid, created_by_user_auth_source
+    from referral r
+    where not exists (select from auth_user where id = r.created_by_userid);
```

Comment:
_not a blocker_
💭 With only local and dev environments, we have the option to avoid doing these data migrations. We could collapse the migrations into a single migration per table before we roll up the pre-prod environment.